create_environment.py

```diff
@@ -0,0 +1,569 @@
+            deps_lock["dependencies"].append(p)
+            recreate = True
+        else:
+
+            installed_version = Version(dependency_specs[package_name][0])
+            if constraint and installed_version not in SpecifierSet(constraint):
+                recreate = True
+                # We need another version
+                deps_lock["dependencies"] = [
+                    d
+                    for d in deps_lock["dependencies"]
+                    if not d.startswith(package_name)
+                ]
+                deps_lock["dependencies"].append(p)
+    if recreate:
+        deps_lock["dependencies"].sort()
+        env_file.write_text(yaml.safe_dump(deps_lock))
+    return recreate
+
+
+def main(input_dir, prefix_dir, force=False):
+    """
+    Main function to create the Conda environment for a tool.
+
+    Parameters
+    ----------
+    input_dir : Path
+        Path to the tool definition directory.
+    prefix_dir : Path
+        Installation prefix for the Conda environment.
+    force : bool, optional
+        Whether to force recreation of the environment, by default False.
+    """
+
+    mamba_url = get_download_url()
+    input_dir = input_dir.expanduser().absolute()
+    config = load_config(input_dir)
+    version = config["tool"].get(
+        "version", config.get("project", {}).get("version")
+    )
+    if not version:
+        raise ValueError("You need to define a version.")
+    env_dir = (
+        prefix_dir.expanduser().absolute()
+        / config["tool"]["name"]
+        / version.lower().strip("v")
+    )
+    if force is True or check_for_environment_creation(
+        input_dir, config["tool"]["run"]["dependencies"]
+    ):
+        with TemporaryDirectory() as temp_dir:
+            tar_path = Path(temp_dir) / "micromamba.tar.bz2"
+            download_with_progress(mamba_url, tar_path)
+            extract_micromamba(tar_path, temp_dir)
+            micromamba_path = Path(temp_dir) / "bin" / "micromamba"
+            if not micromamba_path.is_file:
+                raise ValueError(
+                    "Micromamba binary was not found after extraction."
+                )
+            create_environment(Path(temp_dir), input_dir, env_dir, config)
+            set_version(env_dir, version, new=True)
+    else:
+        set_version(env_dir, version, new=False)
+    share_dir = env_dir / "share" / "tool" / config["tool"]["name"]
+    share_dir.mkdir(exist_ok=True, parents=True)
+    build_env_file = input_dir / "build-environment.lock"
+    copy_all(input_dir, share_dir)
+    try:
+        build(
+            env_dir.parent,
+            share_dir,
+            build_env_file,
+            config["tool"].get("build", {}),
+        )
+    except Exception as error:
+        logger.error(error)
+        shutil.rmtree(env_dir)
+        raise ValueError("Failed to create environment.")
+    print("The tool was successfully deployed in:", share_dir)
```

Comment:
what do u think of taking downloading micromamba out of the if loop, since we need it for build stage as well?
soemthing like this:

```python
    mamba_temp_dir = TemporaryDirectory()
    try:
        tar_path = Path(mamba_temp_dir.name) / "micromamba.tar.bz2"
        download_with_progress(mamba_url, tar_path)
        extract_micromamba(tar_path, mamba_temp_dir.name)
        micromamba_path = Path(mamba_temp_dir.name) / "bin" / "micromamba"
        
        if not micromamba_path.is_file():
            raise ValueError(
                "Micromamba binary was not found after extraction."
            )

        if force is True or check_for_environment_creation(
            input_dir, config["tool"]["run"].get("dependencies", [])
        ):
            create_environment(Path(mamba_temp_dir.name), input_dir, env_dir, config)
            set_version(env_dir, version, new=True)
        else:
            set_version(env_dir, version, new=False)

        share_dir = env_dir / "share" / "tool" / config["tool"]["name"]
        share_dir.mkdir(exist_ok=True, parents=True)
        build_env_file = input_dir / "build-environment.lock"
        copy_all(input_dir, share_dir)

        try:
            build(
                env_dir.parent,
                share_dir,
                build_env_file,
                config["tool"].get("build", {}),
                micromamba_path=micromamba_path
            )
        except Exception as error:
            logger.error(error)
            shutil.rmtree(env_dir)
            raise ValueError("Failed to create environment.")
        print("The tool was successfully deployed in:", share_dir)
    finally:
        mamba_temp_dir.cleanup()
```
And then in the `build` function we need to change the [`mamba`](https://github.com/FREVA-CLINT/data-analysis-tools/blob/add-example-tool/create_environment.py#L380) with `str(micromamba_path)` while we already changed the `build(env_dir, build_dir, env_file, build_config):` to `build(env_dir, build_dir, env_file, build_config, micromamba_path):`